### PR TITLE
[EA] People directory map

### PR DESCRIPTION
**Pull request by oetherington**
Adds a user map to the people directory along with:

- `null` filters in elasticsearch to exclude users without a location set
- adds dark mode styles for map (which also required converting `SmallMapPreview` into a functional component)
- removes the community map from the groups page

Screenshot 2024-05-29 at 17 14 16

Screenshot 2024-05-29 at 17 14 25

Screenshot 2024-05-29 at 17 13 57

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1207433125523225) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by oetherington** [comment]
FYI @agnesstenlund

File: packages/lesswrong/components/hooks/useMapStyle.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import { useConcreteThemeOptions } from "../themes/useTheme";
+
+export const useMapStyle = () => {
+  const themeOptions = useConcreteThemeOptions();
+  const isDarkMode = themeOptions.name === "dark";
+  if (!isFriendlyUI) {
+    return "mapbox://styles/habryka/cilory317001r9mkmkcnvp2ra";
+  }
+  return isDarkMode
+    ? "mapbox://styles/mapbox/dark-v11"
```
Comment (oetherington) [inline packages/lesswrong/components/hooks/useMapStyle.ts:11]: This is just the default dark mode style

File: packages/lesswrong/components/peopleDirectory/PeopleDirectoryResults.tsx [hunk 2]
```diff
@@ -1,99 +1,18 @@
...
-    width: "100%",
-  },
-  loadMore: {
-    gridColumn: "1/-1",
-  },
-});
-
-export const PeopleDirectoryResults = ({classes}: {
-  classes: ClassesType<typeof styles>,
-}) => {
```
Comment (oetherington) [inline packages/lesswrong/components/peopleDirectory/PeopleDirectoryResults.tsx:51]: Current implementation moved into `PeopleDirectoryResultsList`

File: packages/lesswrong/server/search/elastic/ElasticQuery.ts [hunk 3]
```diff
@@ -31,7 +31,7 @@ export type QueryFilter = {
   field: string,
 } & ({
   type: "facet",
-  value: boolean | string,
+  value: boolean | string | null,
```
Comment (oetherington) [inline packages/lesswrong/server/search/elastic/ElasticQuery.ts:34]: Support filtering out `null`s

File: packages/lesswrong/themes/globalStyles/miscStyles.ts [hunk 4]
```diff
@@ -254,4 +255,21 @@ div#mocha {
   width: 20px;
   z-index: 10000000;
 }
+
+/* Mapbox */
```
Comment (oetherington) [inline packages/lesswrong/themes/globalStyles/miscStyles.ts:259]: More dark mode styles

**Comment by agnesstenlund** [comment]
Fantastic 🤩

**Comment by Will-Howard** [comment]
I'm still getting the map in the groups page, did you forget to commit something?
Screenshot 2024-06-14 at 10 18 27

**Comment by Will-Howard** [comment]
Probably not added in this PR but just reporting that I'm getting this scroll arrow on the right when there is no need to scroll
Screenshot 2024-06-14 at 10 21 38

**Review by Will-Howard — APPROVED** [review]
LGTM, apart from removing the map from the groups page

**Comment by oetherington** [comment]
@Will-Howard Thanks! The groups page map will be removed when we actually role out the people directory in #9345. I'll take a look at the eroneous arrow in a separate PR.

**Pull request merged**